buildtest/buildsystem/base.py

```diff
@@ -466,31 +519,82 @@ def run_tests(self, testfile):
-            fd.write("\n".join(out))
-        self.logger.debug(f"Writing run output to file: {run_output_file+'.out'}")
+
+        out = "\n".join(out)
+        err = "\n".join(err)
+
+        write_file(self.metadata["outfile"], out)
+        self.logger.debug(f"Writing run output to file: {self.metadata['outfile']}")
+
         # write error from test to .err file
-        with open(run_output_file + ".err", "w") as fd:
-            fd.write("\n".join(err))
-        self.logger.debug(f"Writing run error to file: {run_output_file + '.err'}")
+        write_file(self.metadata["errfile"], err)
+        self.logger.debug(f"Writing run error to file: {self.metadata['errfile']}")
+
         self.logger.debug(f"Return code: {command.returncode} for test: {testfile}")
         result["RETURN_CODE"] = command.returncode
         result["END_TIME"] = self.get_formatted_time("end_time")
 
-        # Print the test result for the user
-        if command.returncode == 0:
-            print(
-                "{:<30} {:<30} {:<30} {:<30}".format(
-                    self.config_name, self.name, "PASSED", self.buildspec
+        status = self.recipe.get("status")
+
+        test_state = ""
+
+        # if status is defined in Buildspec, then check for returncode and regex
+        if status:
+
+            # returncode_match is boolean to check if reference returncode matches return code from test
+            returncode_match = True
+
+            # regex_match is boolean to check if output/error stream matches regex defined in Buildspec,
+            # if no regex is defined we set this to True since we do a logical AND
+            regex_match = True
+
+            if "returncode" in status:
+                self.logger.debug("Conducting Return Code check")
+                self.logger.debug(
+                    "Status Return Code: %s   Result Return Code: %s"
+                    % (status["returncode"], result["RETURN_CODE"])
                 )
+                # self.check_returncode checks if test returncode matches returncode specified in Buildspec.
+                # The return value is a bool
+                returncode_match = self.check_returncode(
+                    status["returncode"], result["RETURN_CODE"]
+                )
+
+            if "regex" in status.keys():
+                self.logger.debug("Conducting Regular Expression check")
+                # self.check_regex  applies regular expression check specified in Buildspec with output or error
+                # stream. self.check_regex returns a boolean (True/False) by using re.search
+                regex_match = self.check_regex(status["regex"])
+
+            self.logger.info(
+                "ReturnCode Match: %s Regex Match: %s "
+                % (returncode_match, regex_match)
             )
+
+            if returncode_match and regex_match:
+                test_state = "PASS"
+            else:
+                test_state = "FAIL"
```

Comment:
If this is default (FAIL), you don't need the else: fail. It will already be set and you just need to set to be success given the first condition. It's the same for line ~587.

Reply:
okay i believe this is fixed in https://github.com/shahzebsiddiqui/buildtest-framework/commit/b4750fdab1dd4367d448f7c51dd21e3098c3e92d